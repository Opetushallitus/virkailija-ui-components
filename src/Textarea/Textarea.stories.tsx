import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import Textarea from './index';

const stories = storiesOf('Components/Textarea', module);

stories.add('Default', () => {
  const disabled = boolean('Disabled', false) as boolean;
  const error = boolean('Error', false) as boolean;

  return <Textarea disabled={disabled} error={error} placeholder="Textarea" />;
});
