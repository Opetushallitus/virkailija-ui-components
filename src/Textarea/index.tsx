import * as React from 'react';

import { InputBase } from '../Input';

const TextareaBase = InputBase.withComponent('textarea');

type TextareaBaseProps = {
  error?: boolean;
  disabled?: boolean;
};

export type TextareaProps = TextareaBaseProps &
  Omit<React.ComponentProps<typeof TextareaBase>, keyof TextareaBaseProps>;

const Textarea = ({ rows = 3, error = false, disabled = false, ...props }) => (
  <TextareaBase error={error} disabled={disabled} rows={rows} {...props} />
);

export default Textarea;
