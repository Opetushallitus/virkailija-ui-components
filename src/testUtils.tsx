import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createTheme } from './createTheme';

const theme = createTheme();

const Providers = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const dummyDiv = document.createElement('div');
dummyDiv.setAttribute('data-nothing', 'true');

// Return rootElement from render() for easier snapshot testing
const customRender = (ui: React.ReactElement, options: any = {}) => {
  const result = render(ui, { wrapper: Providers, ...options });
  return {
    ...result,
    rootElement: result.container.firstChild || dummyDiv,
  };
};

export * from '@testing-library/react';

export { customRender as render };
