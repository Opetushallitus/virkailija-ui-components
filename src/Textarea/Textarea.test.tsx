import React from 'react';
import { render } from '../testUtils';
import Textarea from './index';

it('Renders correctly with defaults', () => {
  const { rootElement } = render(<Textarea />);
  expect(rootElement).toMatchSnapshot();
});
