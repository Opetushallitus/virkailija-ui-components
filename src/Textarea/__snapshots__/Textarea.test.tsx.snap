// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with defaults 1`] = `
.c0 {
  border: 1px solid #ced4da;
  color: #666;
  font-family: 'Roboto',sans-serif;
  line-height: 1.5;
  border-radius: 4px;
  outline: none;
  padding: 6px 12px;
  font-size: 1rem;
  -webkit-transition: border-color 0.25s,box-shadow 0.25s;
  transition: border-color 0.25s,box-shadow 0.25s;
  display: block;
  width: 100%;
  box-sizing: border-box;
  background-color: white;
  box-shadow: 0 0 0 0 transparent;
}

.c0:hover,
.c0:focus {
  border-color: #0a789c;
}

.c0:focus {
  box-shadow: 0 0 0 3px rgba(10,120,156,0.2);
}

<textarea
  class="c0"
  rows="3"
  type="text"
/>
`;
